ui/src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+const useDebounce = (value, delay) => {
```

Comment:
why can't we use lodash debounce or something else that already exists?

Reply:
its the solution i found to seperate the state update and the filter operation.
I as see it the trade off will be to hold another variable and make the debounce on him and useEffect to listen to that change to make the filter operation.
I come from the assumption, the state must be updated immediately